ipi_tests/examples/modify_xml.py

```diff
@@ -0,0 +1,44 @@
+import xml.etree.ElementTree as ET
+import sys
+
+
+def modify_xml(
+    input_name,
+    output_name,
+    socket_mode="unix",
+    port_number=33333,
+    address_name="localhost",
+    nsteps=5,
+):
+    """ Modify xml to run dummy tests """
+    tree = ET.parse(input_name)
+    root = tree.getroot()
+    clients = list()
+
+    for ffsocket in root.findall("ffsocket"):
+        name = ffsocket.attrib["name"]
+        ffsocket.attrib["mode"] = socket_mode
+
+        for element in ffsocket:
+            port = port_number + len(clients)
+            if element.tag == "port":
+                element.text = str(port)
+            elif element.tag == "address":
+                element.text = address_name + "_" + str(port)
+                address = address_name + "_" + str(port)
+        clients.append((port, address))
+
+    element = root.find("total_steps")
+    if element is not None:
+        element.text = str(nsteps)
+    else:
+        new_element = ET.SubElement(root, "total_steps")
+        new_element.text = str(nsteps)
+
+    tree.write(open(output_name, "wb"))
+
+
+if __name__ == "__main__":
+    name = sys.argv[1]
+    out = "output.xml"
+    modify_xml(name, out)
```

Comment:
Why both this script and the one above (examplestools) are needed?

Reply:
I am not sure if understand the question. I try an answer though.

The section highlighted above allows calling the function modify_xml for debugging
or experimenting purposes.
The exampletools file contains the object definition and functions required
to find all the examples, to check if they are in the excluded list or not, to create the temporary folders, to modify the xml file (change the number of steps to 2, etc)  and run the examples.

Reply:
The address names are changed in such a way that if one test fails, and the ipi_localhost is not erased, the other tests can still run.

Reply:
In principle, it is possible to create a modify_xml function that can deal with all the necessities of examples and regression tests.
However, it is not desirable to have two simultaneous PR that modify the same file. The reason is that unless they are merged without further modification, which in general should not be assumed, conflicts will appear.

One way to proceed is that after the PRs are accepted, one can create an issue to discuss if it is better to entangle the two tasks ( checking examples and regression tests) or it is better to let them evolve independently.

I actually would say that, even though in general cases code repetition is a bad practice, for this particular situation it is better to leave them unentangled to have 2 simple functions instead of having a big one that considers all the possibilities and corner cases and will be more difficult to maintain and modify.

Reply:
I also agree that it is a bit confusing to have two modify_xml files. Perhaps the two functions should have different names.

Reply:
I think different names is good for now too.

Reply:
It seems that my idea of having a different file for testing and experimenting purposes is creating more confusion than what it may help future developers. For now, I think it's better to delete it.